.github/workflows/build.yml

```diff
@@ -0,0 +1,69 @@
+name: Build GetStarted example
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    env:
+        ARM_UBL_ACTIVATION_CODE: ${{secrets.ARM_TOKEN}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+#      - name: Cache vcpkg
+#        uses: actions/cache@v3
+#        with:
+#          key: vcpkg
+#          path: /home/runner/.vcpkg
+
+
+      - name: Cache vcpkg
+        uses: actions/cache@v3
+        with:
+          key: vcpkg-${{ runner.os }}-${{ runner.arch }}-${{ github.run_id }}-${{ matrix.compiler }}
+          restore-keys: |
+            vcpkg-${{ runner.os }}-${{ runner.arch }}-
+          path: /home/runner/.vcpkg
+
+          
+      - name: Activate Arm tool license and build
+        run: |
+          . <(curl https://aka.ms/vcpkg-init.sh -L)
```

Comment:
Consider using my composite action for vcpkg activation:
https://github.com/JonatanAntoni/actions/blob/main/vcpkg/action.yml

For an example look here:
https://github.com/ARM-software/CMSIS-RTX/blob/7b056a242e98587b66eb7c1c3a73ff2ab6cb0539/.github/workflows/cmsis_rv2.yml#L74-L78